### PR TITLE
drivers: bluetooth: Rename Silabs HCI driver

### PR DESCRIPTION
**Pull request by jhedberg**
Rename the Silabs HCI driver to hci_silabs_efr32.c to better indicate what hardware it supports. Also rename the associated devicetree binding and Kconfig options to be consistent with the new driver name.

### CODE REVIEW

**Comment by jhedberg** [comment]
Added the DNM flag, since this actually depends on an update to CMakeLists.txt in the HAL tree. I think it's better to wait for #80148 to be merged first since then it's the main tree `modules/hal_silabs/simplicity_sdk/CMakeLists.txt` that needs updating instead.

File: MAINTAINERS.yml
Comment (jerome-pouiller) [inline MAINTAINERS.yml:3534]: You can drop this line since `drivers/*/*_silabs*` also matches it.

File: drivers/bluetooth/hci/CMakeLists.txt
Comment (jerome-pouiller) [inline drivers/bluetooth/hci/CMakeLists.txt:40]: In one side we have `BT_SILABS_EFR32` and the other side `SILABS_BT_HCI_EFR32`. I suggest to include `HCI` in all or in none of them.

File: drivers/bluetooth/hci/hci_silabs_efr32.c
Comment (jerome-pouiller) [inline drivers/bluetooth/hci/hci_silabs_efr32.c:1]: The functions in this file are prefixed with `slz_`. Maybe we could rename them?

**Review by jerome-pouiller — CHANGES_REQUESTED** [review]
I believe we should rename the `slz_` functions before to merge ths PR.

File: drivers/bluetooth/hci/CMakeLists.txt
Comment (jhedberg) [reply drivers/bluetooth/hci/CMakeLists.txt:40]: > In one side we have `BT_SILABS_EFR32` and the other side `SILABS_BT_HCI_EFR32`. I suggest to include `HCI` in all or in none of them.

The reasoning is based on context. In DTS you're far away from Bluetooth land. In `drivers/bluetooth/hci` you're deep inside it.

File: drivers/bluetooth/hci/hci_silabs_efr32.c
Comment (jhedberg) [reply drivers/bluetooth/hci/hci_silabs_efr32.c:1]: > The functions in this file are prefixed with `slz_`. Maybe we could rename them?

Those functions are private to the driver, and I was trying to keep this PR as minimal as possible. The names can be changed at any point without affecting anything outside of the driver. What would you like the names to be? (I considered doing the rename but then dropped the idea since I couldn't decide on a nice and short prefix)

File: MAINTAINERS.yml
Comment (jerome-pouiller) [reply MAINTAINERS.yml:3534]: If it doesn't, I believe we should replace this line by `drivers/*/*/*_silabs*`.

File: drivers/bluetooth/hci/CMakeLists.txt
Comment (jhedberg) [reply drivers/bluetooth/hci/CMakeLists.txt:40]: Additionally, most HCI driver options don't have HCI in the name. If we add it we should add it to all drivers, and such a change is out of scope for this PR. As for DTS, "HCI" is not clear enough on its own since that concept/acronym exists in more domains than just Bluetooth.

File: drivers/bluetooth/hci/hci_silabs_efr32.c
Comment (jerome-pouiller) [reply drivers/bluetooth/hci/hci_silabs_efr32.c:1]: You're right. I foresee a long discussion. Let's postpone that change.

**Review by jerome-pouiller — DISMISSED** [review]

File: MAINTAINERS.yml
Comment (jhedberg) [reply MAINTAINERS.yml:3534]: Out of scope for this PR, but since HCI drivers are the only kind of HCI driver we have, I think it's worth considering dropping the `hci` subdirectory completely. It'd then also be consistent with the fact that the header file for such drivers is `include/zephyr/drivers/bluetooth.h`.

Comment (asmellby) [inline MAINTAINERS.yml:3534]: This change will likely cause a merge conflict with https://github.com/zephyrproject-rtos/zephyr/pull/80128 - nothing to be done about it until one of the PRs is merged, though, unless we roll the other PR into this one.

**Review by asmellby — DISMISSED** [review]

**Comment by jhedberg** [comment]
> Added the DNM flag, since this actually depends on an update to CMakeLists.txt in the HAL tree. I think it's better to wait for #80148 to be merged first since then it's the main tree `modules/hal_silabs/simplicity_sdk/CMakeLists.txt` that needs updating instead.

Rebased now and DNM label removed.

**Review by asmellby — APPROVED** [review]

**Review by jerome-pouiller — APPROVED** [review]

**Pull request merged**
